packages/victory-group/src/victory-group.js

```diff
@@ -19,150 +21,209 @@ const fallbackProps = {
-      : {};
-    const container = standalone
-      ? this.renderContainer(containerComponent, containerProps)
-      : groupComponent;
-    const events = Wrapper.getAllEvents(props);
-    if (!isEmpty(events)) {
-      return (
-        <VictorySharedEvents
-          container={container}
-          eventKey={eventKey}
-          events={events}
-          externalEventMutations={externalEventMutations}
-        >
-          {newChildren}
-        </VictorySharedEvents>
+  const container = React.useMemo(() => {
+    if (standalone) {
+      const defaultContainerProps = defaults(
+        {},
+        containerComponent.props,
+        containerProps
       );
+      return React.cloneElement(containerComponent, defaultContainerProps);
     }
-    return React.cloneElement(container, container.props, newChildren);
+    return groupComponent;
+  }, [groupComponent, standalone, containerComponent, containerProps]);
+
+  const events = React.useMemo(() => {
+    return Wrapper.getAllEvents(props);
+  }, [props]);
+
+  const previousProps = usePreviousProps();
+
+  React.useEffect(() => {
+    if (props.animate) {
+      setState({
+        nodesShouldLoad: false,
+        nodesDoneLoad: false,
+        animating: true
+      });
+    }
+    // This hook will run once when the component is initialized
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  React.useEffect(() => {
+    if (props.animate) {
+      Wrapper.setAnimationState(previousProps, props);
+    }
+  }, [props, previousProps]);
+
+  if (!isEmpty(events)) {
+    return (
+      <VictorySharedEvents
+        container={container}
+        eventKey={eventKey}
+        events={events}
+        externalEventMutations={externalEventMutations}
+      >
+        {newChildren}
+      </VictorySharedEvents>
+    );
   }
-}
+  return React.cloneElement(container, container.props, newChildren);
+};
+
+VictoryGroup.displayName = "VictoryGroup";
+VictoryGroup.role = "group";
+VictoryGroup.propTypes = {
+  ...CommonProps.baseProps,
+  ...CommonProps.dataProps,
+  children: PropTypes.oneOfType([
+    PropTypes.arrayOf(PropTypes.node),
+    PropTypes.node
+  ]),
+  color: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
+  colorScale: PropTypes.oneOfType([
+    PropTypes.arrayOf(PropTypes.string),
+    PropTypes.oneOf([
+      "grayscale",
+      "qualitative",
+      "heatmap",
+      "warm",
+      "cool",
+      "red",
+      "green",
+      "blue"
+    ])
+  ]),
+  horizontal: PropTypes.bool,
+  offset: PropTypes.number
+};
+
+VictoryGroup.defaultProps = {
+  containerComponent: <VictoryContainer />,
+  groupComponent: <g />,
+  samples: 50,
+  sortOrder: "ascending",
+  standalone: true,
+  theme: VictoryTheme.grayscale
+};
+
+VictoryGroup.expectedComponents = [
+  "groupComponent",
+  "containerComponent",
+  "labelComponent"
+];
+
+VictoryGroup.getChildren = getChildren;
+
+export default React.memo(VictoryGroup, isEqual);
```

Comment:
I'm curious about whether this memoization is worth the potentially expensive equality check

Reply:
I had the same question! Right now, it seems to be saving us about 200ms of render time on each interaction—if we take it out the performance is basically on par with the initial measurements. If we're only calling isEqual once per VictoryGroup component, it seems to have a pretty miminal performance impact. The profiler says it takes around 0.1 ms per function call.

<img width="1013" alt="Screen Shot 2021-08-11 at 2 39 44 PM" src="https://user-images.githubusercontent.com/13334214/129107182-728f1c2a-6d4e-424b-9e75-bfe746ecd7e8.png">

Reply:
Nice!